fix(dev): added persistent loop for authentication

WarpWing opened the pull request:
**Pull Request Description**

Closes #407 

---

**Licensing Certification**

FarmData2 is a [Free Cultural Work](https://freedomdefined.org/Definition) and all accepted contributions are licensed as described in the LICENSE.md file. This requires that the contributor holds the rights to do so. By submitting this pull request **I certify that I satisfy the terms of the [Developer Certificate of Origin](https://developercertificate.org/)** for its contents.

WarpWing committed:
fix(dev): added persistent loop for authentication 

Signed-off-by: Ty Chermsirivatana <chermsit@dickinson.edu>

braughtg reviewed (changes requested):
This has some good elements and ideas in it.  However, it doesn't seem to work correctly.  A few thoughts here:
- The `gh api user` command return 0 when logged in and non-zero when not logged in.  So I think the condition on that needs to not use `!`.  So the loop should be "while not logged in" or `while gh api user`.
- Then in the loop do the `gh auth login` inside the loop.
- Seems like it would be good to move the instructions about doing the login into the loop as well so that they only print if the user is being logged in.

WarpWing committed:
fix(dev): revamped authentication loop

Signed-off-by: Ty Chermsirivatana <chermsit@dickinson.edu>

WarpWing commented:
> This has some good elements and ideas in it. However, it doesn't seem to work correctly. A few thoughts here:
> 
>     * The `gh api user` command return 0 when logged in and non-zero when not logged in.  So I think the condition on that needs to not use `!`.  So the loop should be "while not logged in" or `while gh api user`.
> 
>     * Then in the loop do the `gh auth login` inside the loop.
> 
>     * Seems like it would be good to move the instructions about doing the login into the loop as well so that they only print if the user is being logged in.

Thanks for the feedback, I've implemented your suggestions! I moved the `gh auth login` login inside and make the logic revolve around `gh api user`.

braughtg committed:
prints github login instructions only once

braughtg committed:
Merge branch 'development' into patch-3
